nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -61,4 +64,44 @@ bool Controller::computeVelocityCommand(
   return true;
 }
 
+rcl_interfaces::msg::SetParametersResult
+Controller::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameters)
+{
+  std::lock_guard<std::mutex> lock(dynamic_params_lock_);
```

Comment:
Where's the other lock so that we're locking something?

Reply:
ups